lib/formatters/stringFormatter.js

```diff
@@ -147,6 +146,27 @@ function formatter(messages, source) {
 		output += `${chalk.underline(logFrom(source))}\n`;
 	}
 
+	/**
+	 * @param {import('stylelint').StylelintWarning} message
+	 * @return {string}
+	 */
+	function formatMessageText(message) {
+		let result = message.text;
+
+		result = result
+			// Remove all control characters (newline, tab and etc)
+			.replace(/[\u0001-\u001A]+/g, ' ') // eslint-disable-line no-control-regex
+			.replace(/\.$/, '');
+
+		const ruleString = `(${message.rule})`;
+
+		if (result.endsWith(ruleString)) {
+			result = result.slice(0, result.lastIndexOf(ruleString));
```

Comment:
Thank you! In addition, how about adding `.trim()` as below?

```suggestion
			result = result.slice(0, result.lastIndexOf(ruleString)).trim();
```

```js
// input
'Unexpected invalid hex color "#ggg" (color-no-invalid-hex)'
//=> without trim
'Unexpected invalid hex color "#ggg" '
//=> with trim
'Unexpected invalid hex color "#ggg"'
```

Maybe, we may need more test cases of `stringFormatter.js` to verify this behavior change. 🤔

Reply:
@ybiquitous I added a space to the `ruleString` that we attempt to remove. I also added a test for the behavior.

```diff
-		const ruleString = `(${message.rule})`;
+		const ruleString = ` (${message.rule})`;
```

What do you think?

Reply:
It also looks good! 👍🏼